src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let stateCopy = { ...state };
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        stateCopy = {
+          ...stateCopy, ...action.extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          stateCopy = { ...stateCopy };
+          delete stateCopy[key];
+        });
+        break;
+
+      case 'clear':
+        stateCopy = {};
+        break;
+
+      default:
+        throw new Error('Unknown action type: ' + action.type);
+    }
+
+    states.push(stateCopy);
```

Comment:
you should spread stateCopy
```suggestion
    states.push({ ...stateCopy });
```